package/src/state-store/audio-player.ts

```diff
@@ -0,0 +1,372 @@
+    });
+    if (!this.playerRef) {
+      return;
+    }
+    if (this.playerRef?.setRateAsync) {
+      await this.playerRef.setRateAsync(nextPlaybackRate, true, 'high');
+    }
+  }
+
+  play() {
+    if (this.isPlaying) {
+      return;
+    }
+
+    if (this._pool) {
+      this._pool.requestPlay(this.id);
+    }
+
+    if (this.previewVoiceRecording) {
+      if (NativeHandlers.Audio?.resumePlayer) {
+        NativeHandlers.Audio.resumePlayer();
+      }
+      this.state.partialNext({
+        isPlaying: true,
+      });
+      return;
+    }
+
+    if (!this.playerRef) {
+      return;
+    }
+
+    if (this.isExpoCLI) {
+      if (this.playerRef?.playAsync) {
+        this.playerRef.playAsync();
+      }
+    } else {
+      if (this.playerRef?.resume) {
+        this.playerRef.resume();
+      }
+    }
+    this.state.partialNext({
+      isPlaying: true,
+    });
+  }
+
+  pause() {
+    if (!this.isPlaying) {
+      return;
+    }
+    if (this.previewVoiceRecording) {
+      if (NativeHandlers.Audio?.pausePlayer) {
+        NativeHandlers.Audio.pausePlayer();
+      }
+      this.state.partialNext({
+        isPlaying: false,
+      });
+      return;
+    }
+
+    if (!this.playerRef) {
+      return;
+    }
+
+    if (this.isExpoCLI) {
+      if (this.playerRef?.pauseAsync) {
+        this.playerRef.pauseAsync();
+      }
+    } else {
+      if (this.playerRef?.pause) {
+        this.playerRef.pause();
+      }
+    }
+    this.state.partialNext({
+      isPlaying: false,
+    });
+
+    if (this._pool) {
+      this._pool.notifyPaused();
+    }
+  }
+
+  toggle() {
+    if (this.isPlaying) {
+      this.pause();
+    } else {
+      this.play();
+    }
+  }
+
+  async seek(positionInSeconds: number) {
+    if (this.previewVoiceRecording) {
+      this.position = positionInSeconds;
+      if (NativeHandlers.Audio?.seekToPlayer) {
+        NativeHandlers.Audio.seekToPlayer(positionInSeconds * 1000);
+      }
+      return;
+    }
+    if (!this.playerRef) {
+      return;
+    }
+    this.position = positionInSeconds;
+    if (this.isExpoCLI) {
+      if (positionInSeconds === 0) {
+        // If currentTime is 0, we should replay the video from 0th position.
+        if (this.playerRef?.replayAsync) {
+          await this.playerRef.replayAsync({});
+        }
+      } else {
+        if (this.playerRef?.setPositionAsync) {
+          await this.playerRef.setPositionAsync(positionInSeconds);
+        }
+      }
+    } else {
+      if (this.playerRef?.seek) {
+        this.playerRef.seek(positionInSeconds);
+      }
+    }
+  }
+
+  async stop() {
+    // First seek to 0 to stop the audio and then pause it
+    await this.seek(0);
+    this.pause();
+  }
+
+  onRemove() {
+    if (this.previewVoiceRecording) {
+      if (NativeHandlers.Audio?.stopPlayer) {
+        NativeHandlers.Audio.stopPlayer();
+      }
+      this.state.partialNext({
+        ...INITIAL_STATE,
+        currentPlaybackRate: this.playbackRates[0],
```

Comment:
*nitpick*

Why are these simply not part of the `INITIAL_STATE` ?

Reply:
How is including them into `INITIAL_STATE` going to prevent you from overriding them ? Or what is the purpose of overriding them if we are overriding it just back ?

Reply:
The `playbackRates` is in the options as well for the ease of overriding the value when we instantiate a new player as its mostly gonna be constant across the player so we override it through the options but actually set it to the store internally here in the lines above. Its not about not being able to override the playbackRates through state but just ease of overriding through options as its more of a stable option for the class AudioPlayer and no one would update the state for it on the whim.